### PR TITLE
librbd/cache/pwl: fix parsing of cache_type in create_image_cache_state()

### PR DESCRIPTION
**Pull request by idryomov**
cache_type is a string (currently "ssd" or "rwl"), not an int.  When
coerced to an int, the result is 0 -- JSONFormattable ends up calling
atoi() under the hood and atoi() does not detect errors...

As the value of rbd_persistent_cache_mode option is validated by the
option parser (enum type) and, separately, ImageCacheState::cache_type
is validated in InitRequest::get_image_cache_state() when creating
an instance of the approprate WriteLog, don't attempt to validate it
in ImageCacheState::create_image_cache_state().

Fixes: https://tracker.ceph.com/issues/50668
Signed-off-by: Ilya Dryomov <idryomov@gmail.com>

### CODE REVIEW

**Comment by idryomov** [comment]
cc @MahatiC

**Comment by idryomov** [comment]
cc @CongMinYin

**Comment by MahatiC** [comment]
lgtm, thanks!

File: src/librbd/cache/pwl/ImageCacheState.cc [hunk 1]
```diff
@@ -141,19 +143,10 @@ ImageCacheState<I>* ImageCacheState<I>::create_image_cache_state(
     }
 
     bool cache_exists = (bool)f["present"];
-    int cache_type = (int)f["cache_type"];
-
-    switch (cache_type) {
-      case IMAGE_CACHE_TYPE_SSD:
```
Comment (CongMinYin) [inline src/librbd/cache/pwl/ImageCacheState.cc:147]: Hi @MahatiC ，here I have a question. Why didn't here create a `cache_state` when the type was `IMAGE_CACHE_TYPE_SSD`, was this a error before?

Comment (MahatiC) [reply src/librbd/cache/pwl/ImageCacheState.cc:147]: @CongMinYin 
```
case IMAGE_CACHE_TYPE_SSD:
case IMAGE_CACHETYPE_RWL:
// create cache_state
```
That syntax means, for both cases, create the `cache_state`. You can refer to the syntax details here[0]

[0]https://www.learncpp.com/cpp-tutorial/switch-fallthrough-and-scoping/

Comment (CongMinYin) [reply src/librbd/cache/pwl/ImageCacheState.cc:147]: > @CongMinYin
> 
> ```
> case IMAGE_CACHE_TYPE_SSD:
> case IMAGE_CACHETYPE_RWL:
> // create cache_state
> ```
> 
> That syntax means, for both cases, create the `cache_state`. You can refer to the syntax details here[0]
> 
> [0]https://www.learncpp.com/cpp-tutorial/switch-fallthrough-and-scoping/

Sorry for my misunderstanding.

**Comment by CongMinYin** [comment]
lgtm.

**Pull request merged**